src/consts/Stages.ts

```diff
@@ -184,38 +193,115 @@ export const STAGES: ReadonlyArray<STAGE> = [
         readyForNext: () => true,
         previous: undefined,
         previousEventTitle: undefined,
+        showStage: () => true,
       },
       {
         key: "SelectionPackageOSFApproval",
         text: "OSF Approval",
-        next: () => "PackageApproval",
-        nextEventTitle: () =>
-          "Forward Stage Change: OSF Approval to Package Approval",
+        next: (request) => {
+          if (request?.csfcaApproval === "Yes") {
+            return "SelectionPackageCSFApproval";
+          } else {
+            return STAGES.find(({ key }) => key === "PackageApproval")
+              ?.subStages?.find(
+                ({ key }) => key === "SelectionPackageCSFApproval"
+              )
+              ?.next?.(request);
+          }
+        },
+        nextEventTitle: (request) => {
+          const next = STAGES.find(({ key }) => key === "PackageApproval")
+            ?.subStages?.find(
+              ({ key }) => key === "SelectionPackageOSFApproval"
+            )
+            ?.next?.(request);
+          return `Forward Stage Change: OSF Approval to ${next}`;
+        },
```

Comment:
This works great! My only feedback would be that it logs the key value, so might be nice to take the key, and lookup the title, to log the pretty name like other event titles.

Reply:
Great update @RobertPorterfield!